keras_cv/models/convnext.py

```diff
@@ -0,0 +1,494 @@
+            f"`pooling` must be `None` when `include_top=True`."
+            f"Received pooling={pooling} and include_top={include_top}. "
+        )
+
+    inputs = utils.parse_model_inputs(input_shape, input_tensor)
+
+    x = inputs
+    if include_rescaling:
+        x = layers.Rescaling(1 / 255.0)(x)
+
+    # Stem block.
+    stem = keras.Sequential(
+        [
+            layers.Conv2D(
+                projection_dims[0],
+                kernel_size=4,
+                strides=4,
+                name=model_name + "_stem_conv",
+            ),
+            layers.LayerNormalization(
+                epsilon=1e-6, name=model_name + "_stem_layernorm"
+            ),
+        ],
+        name=model_name + "_stem",
+    )
+
+    # Downsampling blocks.
+    downsample_layers = []
+    downsample_layers.append(stem)
+
+    num_downsample_layers = 3
+    for i in range(num_downsample_layers):
+        downsample_layer = keras.Sequential(
+            [
+                layers.LayerNormalization(
+                    epsilon=1e-6,
+                    name=model_name + "_downsampling_layernorm_" + str(i),
+                ),
+                layers.Conv2D(
+                    projection_dims[i + 1],
+                    kernel_size=2,
+                    strides=2,
+                    name=model_name + "_downsampling_conv_" + str(i),
+                ),
+            ],
+            name=model_name + "_downsampling_block_" + str(i),
+        )
+        downsample_layers.append(downsample_layer)
+
+    # Stochastic depth schedule.
+    # This is referred from the original ConvNeXt codebase:
+    # https://github.com/facebookresearch/ConvNeXt/blob/main/models/convnext.py#L86
+    depth_drop_rates = [float(x) for x in np.linspace(0.0, drop_path_rate, sum(depths))]
+
+    # First apply downsampling blocks and then apply ConvNeXt stages.
+    cur = 0
+
+    num_convnext_blocks = 4
+    for i in range(num_convnext_blocks):
+        x = downsample_layers[i](x)
+        for j in range(depths[i]):
+            x = ConvNeXtBlock(
+                projection_dim=projection_dims[i],
+                drop_path_rate=depth_drop_rates[cur + j],
+                layer_scale_init_value=layer_scale_init_value,
+                name=model_name + f"_stage_{i}_block_{j}",
+            )(x)
+        cur += depths[i]
+
+    if include_top:
+        x = Head(
+            num_classes=classes, activation=classifier_activation, name=model_name
+        )(x)
+
+    else:
+        if pooling == "avg":
+            x = layers.GlobalAveragePooling2D()(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D()(x)
+        x = layers.LayerNormalization(epsilon=1e-6)(x)
+
+    model = keras.Model(inputs=inputs, outputs=x, name=model_name)
+
+    if weights is not None:
+        model.load_weights(weights)
+
+    return model
+
+
+def ConvNeXtTiny(
+    include_rescaling,
+    drop_path_rate,
+    layer_scale_init_value,
+    include_top,
```

Comment:
For API consistency, please make `include_rescaling` and `include_top` the first two required/positional parameters to each public ConvNeXt method